modules/sagemaker/sagemaker-model-package-group/sagemaker_model_package_group/stack.py

```diff
@@ -0,0 +1,212 @@
+        construct_id
+            The construct ID of this stack
+        model_package_group_name
+            The Model Package Group name.
+        retain_on_delete, optional
+            Wether or not to retain resources on delete. Defaults True.
+        target_event_bus_arn, optional
+            The event bus arn in to send events model package group state change events to.
+            It can be a bus located in another account. Defaults None.
+        model_package_group_description, optional
+            The model package group description. Defaults None.
+        target_account_ids, optional
+            The target account ids which shall have read-only access to the model package group. Defaults None.
+        sagemaker_project_id, optional
+            SageMaker project id, defaults None
+        sagemaker_project_name, optional
+            SageMaker project name, defaults None
+        """
+        super().__init__(scope, construct_id, **kwargs)
+        self.target_event_bus_arn = target_event_bus_arn
+        self.target_account_ids = target_account_ids
+        self.model_package_group_name = model_package_group_name
+
+        self.removal_policy = (
+            RemovalPolicy.RETAIN if retain_on_delete else RemovalPolicy.DESTROY
+        )
+
+        self.model_package_group_description = model_package_group_description
+        self.sagemaker_project_name = sagemaker_project_name
+        self.sagemaker_project_id = sagemaker_project_id
+
+        self.setup_resources()
+
+        self.setup_outputs()
+
+        self.setup_tags()
+
+    def setup_resources(self) -> None:
+        """Deploy resources."""
+
+        self.model_package_group = self.setup_model_package_group()
+
+        self.event_rule = self.setup_events()
+
+    def setup_model_package_group(self) -> sagemaker.CfnModelPackageGroup:
+        """Create a Model Package Group."""
+        model_package_group_policy = self.get_model_package_group_resource_policy()
+
+        model_package_group = sagemaker.CfnModelPackageGroup(
+            self,
+            "ModelPackageGroup",
+            model_package_group_name=self.model_package_group_name,
+            model_package_group_description=self.model_package_group_description,
+            model_package_group_policy=model_package_group_policy,
+        )
+
+        model_package_group.apply_removal_policy(self.removal_policy)
+
+        return model_package_group
+
+    def get_model_package_group_resource_policy(self) -> Optional[Dict[str, Any]]:
+        """Get a resource policy to enable cross account access into Sagemaker Model Package Group.
+
+        Returns
+        -------
+            A resource policy with cross-account read-only permissions.
+        """
+        if not self.target_account_ids:
+            return None
+
+        sagemaker_arn = f"arn:aws:sagemaker:{self.region}:{self.account}"
+        target_accounts = [f"arn:aws:iam::{a}:root" for a in self.target_account_ids]
```

Comment:
We should use `self.partition` here to make the templates compatible with the China and US GOV partitions.